modules/mesh-task/templates/consul_client_command.tpl

```diff
@@ -26,10 +26,37 @@ login() {
       -token-sink-file /consul/client-token
 }
 
+read_token_stale() {
+    consul acl token read -http-addr ${ consul_http_addr } \
+    %{ if tls ~}
+      -ca-file /consul/consul-ca-cert.pem \
+    %{ endif ~}
+      -stale -self -token-file /consul/client-token
+}
+
+# Retry in order to login successfully.
 while ! login; do
     sleep 2
 done
 
+# Wait for raft replication to hopefully occur. Without this, an "ACL not found" may be cached for a while.
+# Technically, the problem could still occur but this should handle most cases.
+# This waits at most 2s (20 attempts with 0.1s sleep)
+COUNT=20
+while [ "$COUNT" -gt "0" ]; do
+    echo "Checking that the ACL token exists when reading it in the stale consistency mode ($COUNT attempts remaining)"
+    if read_token_stale; then
+        echo "Successfully read ACL token from the server"
+        break
+    fi
+    sleep 0.1
+    COUNT=$((COUNT - 1))
```

Comment:
nit: can just use `((COUNT--))` here

Reply:
Actually, the Consul image is Alpine-based, and doesn't include bash. It's just a posix shell (BusyBox shell, which is `ash`) and it doesn't seem to have the decrement operation.